OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -767,20 +773,19 @@ private async Task<Expression<Func<Workshop, bool>>> ApplyFilterForRegionAdmin(W
             throw new InvalidOperationException(errorMsg);
         }
 
-        filter.InstitutionId = regionAdmin.InstitutionId;
+        var predicate = PredicateBuilder.True<Workshop>();
+        var institutionId = regionAdmin.InstitutionId;
+        var catottgId = regionAdmin.CATOTTGId;
+
+        predicate = predicate.And(x => x.InstitutionHierarchy.InstitutionId == institutionId);
 
         var subSettlementsIds = await codeficatorService
-            .GetAllChildrenIdsByParentIdAsync(regionAdmin.CATOTTGId)
+            .GetAllChildrenIdsByParentIdAsync(catottgId)
             .ConfigureAwait(false);
 
-        var tempPredicate = PredicateBuilder.False<Workshop>();
+        predicate = predicate.And(x => subSettlementsIds.Contains(x.Address.CATOTTGId));
 
-        foreach (var item in subSettlementsIds)
-        {
-            tempPredicate = tempPredicate.Or(x => x.Address.CATOTTGId == item);
-        }
-
-        return tempPredicate;
+        return predicate;
```

Comment:
could you please reorder the code so the predicate will be built and returned in single line ?